sound_byte/lib/pages/friendList.dart

```diff
@@ -4,4 +4,99 @@ import 'package:sound_byte/services/authentication.dart';
 import 'package:sound_byte/pages/userProfile.dart';
 import 'package:sound_byte/pages/friendProfile.dart';
 import 'package:sound_byte/pages/friendScreen.dart';
+import 'package:sound_byte/pages/user.dart';
 
+class FriendList extends StatelessWidget{
+  User user;
+  List<String> user_friends;
+  int number_of_friends;
+  String user_id;
+
+  FriendList (User user) {
+    this.user = user;
+    this.user_friends = user.friends;
+    this.number_of_friends = user.friends.length();
+    this.user_id = user.userId;
+  }
+
+  FriendList.nullFriendList() {
+    this.user = new User("Sound Byte", 
+                        "Soundbyte@gmail.com", 
+                        "12345");
+    this.user_friends = new List<String>();
+    this.number_of_friends = 0;
+    this.user_id = "12345";
+  }
```

Comment:
Null object pattern should make a null-like object. Use the null User constructor and an empty user_id. Currently it contains a bunch of data that is not conveyed by the name.

Reply:
The null-like object is a method that is called on the off chance that an object is not initialized. This is sort of like a default constructor except it only runs when no constructor (even default) is not called.